crates/compiler/gen_dev/src/generic64/storage.rs

```diff
@@ -1155,28 +1174,43 @@ impl<
         {
             // Claim a location for every join point parameter to be loaded at.
             // Put everything on the stack for simplicity.
-            match *layout {
-                single_register_layouts!() => {
-                    let base_offset = self.claim_stack_size(8);
-                    self.symbol_storage_map.insert(
-                        *symbol,
-                        Stack(Primitive {
-                            base_offset,
-                            reg: None,
-                        }),
-                    );
-                    self.allocation_map
-                        .insert(*symbol, Rc::new((base_offset, 8)));
-                }
-                _ => {
-                    let stack_size = layout_interner.stack_size(*layout);
-                    if stack_size == 0 {
-                        self.symbol_storage_map.insert(*symbol, NoData);
-                    } else {
-                        self.claim_stack_area(symbol, stack_size);
+            let mut layout = *layout;
+
+            'inner: loop {
```

Comment:
Up to you, but I think this might read better as a recursive function. same with loading below.